config/initializers/fips.rb

```diff
@@ -0,0 +1,53 @@
+require "openssl"
+require "digest"
+require "digest/md5"
+require "digest/sha1"
+require 'sprockets'
+require 'openid_connect'
+
+# Suppress warning messages
+original_verbose, $VERBOSE = $VERBOSE, nil
+
+# Remove pre-existing constants if they do exist to reduce the amount of log spam and warnings
+Digest.send(:remove_const, "MD5") if Digest.const_defined?("MD5")
+Digest.const_set("MD5", OpenSSL::Digest::MD5)
+Digest.send(:remove_const, "SHA1") if Digest.const_defined?("SHA1")
+Digest.const_set("SHA1", OpenSSL::Digest::SHA1)
+Digest.send(:remove_const, "SHA256") if Digest.const_defined?("SHA256")
+Digest.const_set("SHA256", OpenSSL::Digest::SHA256)
+
+# override the default Digest with OpenSSL::Digest
+Digest::SHA256 = OpenSSL::Digest::SHA256
+Digest::SHA1 = OpenSSL::Digest::SHA1
+
+# Activate warning messages again
+$VERBOSE = original_verbose
+
+# enable FIPS mode
+OpenSSL.fips_mode = true
+
+# each of the following 3rd party overridden is required since a non FIPS complaint encryption method is used
+# if a non-complaint FIPS method like MD5 is used or a direct use of Digest::encryption-method
+#  (rather than OpenSSL::Digest::encryption-method) is performed
+#  the server will crush on run time
+
+# override ActiveSupport hash_digest_class with FIPS complaint method
+ActiveSupport::Digest.hash_digest_class = OpenSSL::Digest::SHA1.new
+
+# override Sprockets digest_class with FIPS complaint method
+Sprockets::DigestUtils.module_eval do
+  def digest_class
+    OpenSSL::Digest::SHA256
+  end
+end
+Sprockets.config = Sprockets.config.merge(
+    digest_class: OpenSSL::Digest::SHA256
+).freeze
+
+# override OpenIDConnect cache_key with FIPS complaint method
+OpenIDConnect::Discovery::Provider::Config::Resource.module_eval do
+  def cache_key
+    sha256 = Digest::SHA256.hexdigest host
+    "swd:resource:opneid-conf:#{sha256}"
+  end
+end
```

Comment:
These `module_eval`s make me very nervous.  Monkey patching 3rd party classes in an initializer seems quite dangerous:
1. It's brittle.  We're making assumptions about the classes and how they use these methods, presumably based on their current implementation details.   Even assuming you have verified that their current code uses this method for making _all_ decisions about which digest to use, the class is making no public guarantees that this will continue to be true.  Therefore: this behavior may change at any time, even in a minor version update.
2. This could create some very confusing situations for future developers who don't realize that this class has been monkey-patched.

Reply:
We left with ActiveSupport and with OpenIDConnect encryption method overridden.
I agree that it has limits and problems but these 3rd parties don't supply FIPS complaint gems so this solution is kind of a compromise.
Is there another alternative that can be implemented?

Reply:
It might be worth making a pull request against the [openid_connect](https://github.com/nov/openid_connect) repo to see if the author wants to make it FIPS compliant. Looks like it's under pretty active development so it might not take too long.

If that doesn't work, maybe we can fork the repo?

Reply:
Open an issue https://github.com/nov/openid_connect/issues/52

Reply:
openid_connect PR was merged to master

Reply:
openid_connect next version will consist the change

Reply:
@nahumcohen can you please schedule a task to upgrade this version?  I don't want to lose this!

Reply:
The change is in the new openid_connect 1.2.0 that is already released.
We will consume the version with the fix. :)

Reply:
And of course the piece of code is removed. Will be pushed ASAP we have a green branch.